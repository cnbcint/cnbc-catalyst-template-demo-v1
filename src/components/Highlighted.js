import React, { useContext } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { LayoutContext } from "./Main";
import { getMargin } from "../helpers/Layout";
import useAnimation from "../hooks/useAnimation";

const Container = styled.div`
  font-weight: bold;
  font-size: 110%;
`;
const Divider = styled.hr`
  margin: ${getMargin(1)} 0;
  border-color: ${(props) => props.color};
`;

export default function Highlighted(props) {
  const layout = useContext(LayoutContext);
  const { ref: refDevider } = useAnimation({
    animationName: "fadeIn",
    duration: 500,
    delay: 125,
    loop: true,
    autoStart: true,
  });
  const { ref: refContent } = useAnimation({
    animationName: "fadeInLeft",
    duration: 500,
    delay: 0,
    loop: true,
    autoStart: true,
  });
  return (
    <Container className={props.className} style={props.style}>
      <div ref={refContent}>{props.children}</div>
      <Divider ref={refDevider} color={layout.colors[props.dividerColor]} />
    </Container>
  );
}

Highlighted.defaultProps = {
  dividerColor: "primary",
  className: "",
  style: {},
};
Highlighted.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  style: PropTypes.object,
  dividerColor: PropTypes.oneOf(["primary", "secondary", "tertiary"]),
};
