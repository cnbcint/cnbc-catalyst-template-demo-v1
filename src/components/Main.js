import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
