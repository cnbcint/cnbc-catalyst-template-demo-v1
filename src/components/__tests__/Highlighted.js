import React from "react";
import Highlighted from "../Highlighted";
import Main from "../Main";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};
beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});

describe("Highlighted", () => {
  it("Renders correctly", () => {
    const { container } = render(
      <Main {...Theme}>
        <Highlighted>Testing Content</Highlighted>
      </Main>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the correct content", () => {
    const { container } = render(
      <Main {...Theme}>
        <Highlighted>Testing Content</Highlighted>
      </Main>
    );
    expect(container.firstChild.textContent).toMatch("Testing Content");
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
