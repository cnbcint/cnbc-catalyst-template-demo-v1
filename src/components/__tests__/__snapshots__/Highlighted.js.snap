// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlighted Renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: black;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

.c1 {
  font-weight: bold;
  font-size: 110%;
}

.c2 {
  margin: 1rem 0;
  border-color: orange;
}

@media (min-width:768px) {
  .c0 hr {
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="pre-animated"
    >
      Testing Content
    </div>
    <hr
      class="c2 pre-animated"
      color="orange"
    />
  </div>
</div>
`;
